pkg/registration/system_information.go

```diff
@@ -1,5 +1,61 @@
 package registration
 
+import "fmt"
+
 // SystemInformation encapsulates any data which should be stored as system meta information.
 // This can be sockets, vCPUs, memory or any other useful information.
 type SystemInformation = map[string]any
+
+// Indicates there is no system information available when registering.
+// Example:
+//
+//	registration.Register(conn, "hostname", "regcode", NoSystemInformation, NoExtraData)
+//
+// Note: If possible make sure to include system information whenever possible
+var NoSystemInformation = map[string]any{}
+
+// ExtraData encapsulates additional data which might be required to be sent along with the
+// request. This can be cloud instance data or online times of the system, depending on the
+// requirements.
+// Available ExtraData keys are:
+//   - instance data (string): Information provided by a cloud instance (Cloud)
+//   - namespace (string): Namespace in which the API operates (SMT)
+//   - online_at ([]string): OnlineAt definition which records hourly usage time of the system (Cloud)
+type ExtraData = map[string]any
+
+// Use NoExtraData when no extra information is used by the registration or status method.
+// Example:
+//
+//	registration.Register(conn, "regcode", "hostname", NoSystemInformation, NoExtraData)
+var NoExtraData = map[string]any{}
+
+type requestWithInformation struct {
```

Comment:
I created a generic structure here to reuse it both on `register` and `status/keepalive`. So we can just plug this in whenever we need to transport the `system_information` and `extra_data` combo.